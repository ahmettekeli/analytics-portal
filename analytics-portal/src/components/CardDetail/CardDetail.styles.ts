import styled from "styled-components";
import { colors } from "configs";

type PropType = { isActive: boolean };

export const Wrapper = styled.div<PropType>`
  display: flex;
  justify-content: space-around;
  margin: 0.5rem; //TODO 0.5 constan gelebilir. gap.sm vs .
  opacity: ${(props) => (props.isActive ? 1 : 0.5)};

  @media (max-width: 630px) {
    flex-direction: column;
  }
`;

export const DetailElement = styled.div`
  margin: 0.5rem 1rem 0 1rem;
  line-height: 0.7;

  p {
    margin: 0.5rem;
  }

  @media (max-width: 630px) {
    margin: 0px;
    line-height: 1;
  }
`;

export const NumberContainer = styled.p`
  font-weight: bold;
  color: ${colors.cardText};
`;
