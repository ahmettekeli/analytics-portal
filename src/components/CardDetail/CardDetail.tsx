import * as S from "./CardDetail.styles";

function CardDetail({
  isActive,
  campaignCount,
  avgInstalls,
  avgRevenue,
}: {
  isActive: boolean;
  campaignCount: number;
  avgInstalls: number;
  avgRevenue: number;
}) {
  return (
    <S.Wrapper isActive={isActive} data-testid="card-detail">
      <S.DetailElement>
        <S.NumberContainer>{campaignCount}</S.NumberContainer>
        <p>Campaigns</p>
      </S.DetailElement>
      <S.DetailElement>
        <S.NumberContainer>{avgInstalls}</S.NumberContainer>
        <p>Avg. Installs</p>
      </S.DetailElement>
      <S.DetailElement>
        <S.NumberContainer>{avgRevenue}</S.NumberContainer>
        <p>Avg. Revenue</p>
      </S.DetailElement>
    </S.Wrapper>
  );
}

export default CardDetail;
