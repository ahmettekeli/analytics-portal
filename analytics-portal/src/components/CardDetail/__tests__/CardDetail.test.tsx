import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import CardDetail from "../CardDetail";

describe("CardDetail", () => {
  let renderResult: ReturnType<typeof render>;
  const isActive = true;
  const campaignCount = 6;
  const avgInstalls = 53.5;
  const avgRevenue = 107.5;
  beforeEach(() => {
    renderResult = render(
      <CardDetail
        isActive={isActive}
        campaignCount={campaignCount}
        avgInstalls={avgInstalls}
        avgRevenue={avgRevenue}
      />
    );
  });
  afterEach(cleanup);

  test("Renders CardDetail component", () => {
    expect(renderResult.getByTestId("card-detail")).toBeInTheDocument();
  });

  test("Should show campaign count correctly", () => {
    expect(
      renderResult.getAllByText("Campaigns")[0].previousSibling
    ).toHaveTextContent(campaignCount.toString());
  });
  test("Should show Avg. Installs and Avg. Revenue correctly", () => {
    expect(
      renderResult.getAllByText("Avg. Installs")[0].previousSibling
    ).toHaveTextContent(avgInstalls.toString());
    expect(
      renderResult.getAllByText("Avg. Revenue")[0].previousSibling
    ).toHaveTextContent(avgRevenue.toString());
  });

  test("CardDetail matches snapshot", () => {
    const tree = renderer
      .create(
        <CardDetail
          isActive={isActive}
          campaignCount={campaignCount}
          avgInstalls={avgInstalls}
          avgRevenue={avgRevenue}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
