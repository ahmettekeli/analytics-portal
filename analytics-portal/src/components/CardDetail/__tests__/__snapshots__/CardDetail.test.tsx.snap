// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetail CardDetail matches snapshot 1`] = `
<div
  className="sc-bdvvtL Tcmbc"
  data-testid="card-detail"
>
  <div
    className="sc-gsDKAQ dTdkyu"
  >
    <p
      className="sc-dkPtRN bnrKRQ"
    >
      6
    </p>
    <p>
      Campaigns
    </p>
  </div>
  <div
    className="sc-gsDKAQ dTdkyu"
  >
    <p
      className="sc-dkPtRN bnrKRQ"
    >
      53.5
    </p>
    <p>
      Avg. Installs
    </p>
  </div>
  <div
    className="sc-gsDKAQ dTdkyu"
  >
    <p
      className="sc-dkPtRN bnrKRQ"
    >
      107.5
    </p>
    <p>
      Avg. Revenue
    </p>
  </div>
</div>
`;
